view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGeneratorLauncher.java

```diff
@@ -91,4 +97,19 @@ public String getJobConfigKey() {
   private Config getJobConfig(Map<String, Object> properties) {
     return (Config) properties.get(getJobConfigKey());
   }
+
+  private AvroFieldValuePartitioner<GenericRecord> getPartitioner(
+      Map<String, Object> properties, String topic) {
+    AvroFieldValuePartitioner<GenericRecord> partitioner = null;
```

Comment:
Prefer returning directly rather than reassigning. I would also encourage using optional rather than null to avoid coupling this method to where it's called.

Reply:
fixed